tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,214 @@
+    }
+
+    while ($times && console('svirt')->run_cmd($cmd)) {
+        $times--;
+    }
+}
+
+sub login_vm_console {
+    reset_consoles;
+    select_console('sut');
+    assert_screen('grub2', 90);
+    send_key 'ret';
+    select_console('root-console', timeout => 180);
+}
+
+sub set_guest_time {
+    my $n_hrs  = (sprintf "%02d", rand(24));
+    my $n_min  = (sprintf "%02d", rand(59));
+    my $n_sec  = (sprintf "%02d", rand(59));
+    my $g_time = script_output("date -u -s $n_hrs:$n_min:$n_sec +'\%F \%T'");
+
+    return $g_time;
+}
+
+sub get_timelc {
+    my ($time, $timelc, $sec, $min, $hour, $mday, $mon, $year);
+    $time = $_[0];
+
+    my @time = reverse(split /-|:|\s|\//, $time);
+    foreach (@time) {
+        s/^0//;
+        $timelc .= $_ . ",";
+    }
+    $timelc =~ s/,$//;
+    ($sec, $min, $hour, $mday, $mon, $year) = split /,/, $timelc;
+    $year -= 1900;
+    $mon  -= 1;
+    $timelc = timelocal($sec, $min, $hour, $mday, $mon, $year);
+
+    return $timelc;
+}
+
+sub do_power_management_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    record_info('Power Manangement Tests');
+    select_console('svirt');
+
+    # Guest Power Off
+    record_info('Guest Power Off');
+    check_power_state($vm_id, $vm_ip, 'power.off', 0, $VM_POWERED_ON, $VM_POWERED_OFF);
+
+    # Guest Power On
+    record_info('Guest Power On');
+    check_power_state($vm_id, $vm_ip, 'power.on', 1, $VM_POWERED_OFF, $VM_POWERED_ON);
+
+    # Guest Reset
+    record_info('Guest Power Reset');
+    check_power_state($vm_id, $vm_ip, 'power.reset', 1, $VM_POWERED_ON, $VM_POWERED_ON);
+
+    # Guest Restart
+    record_info('Guest Power Restart');
+    check_power_state($vm_id, $vm_ip, 'power.restart', 1, $VM_POWERED_ON, $VM_POWERED_ON);
+
+    # Guest Shutdown
+    record_info('Guest Power Shutdown');
+    check_power_state($vm_id, $vm_ip, 'power.shutdown', 0, $VM_POWERED_ON, $VM_POWERED_OFF);
+}
+
+sub do_networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    record_info('Networking Tests');
+    check_power_state($vm_id, $vm_ip, 'power.on', 1, $VM_POWERED_OFF, $VM_POWERED_ON);
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    login_vm_console();
+
+    assert_script_run('ping -c3 openqa.suse.de');
+    assert_script_run('ping -c3 www.suse.com');
+}
+
+sub do_clock_sync_tests {
+    my ($vm_name, $vm_id, $vm_ip) = @_;
+    my ($g_datetime, $h_datetime, $g_difftime);
+
+    record_info('Clock Sync Tests');
+    check_power_state($vm_id, $vm_ip, 'power.on', 1, $VM_POWERED_OFF, $VM_POWERED_ON);
+
+    select_console('sut') if (current_console() ne 'sut');
+
+    # Set the last day of host time within guest before enabling timesynce
+    $g_datetime = set_guest_time();
+    record_info('Guest current time before syncing up', $g_datetime);
+
+    # Enable timesync service in guest VM
+    if (script_output('vmware-toolbox-cmd timesync status', proceed_on_failure => 1, timeout => 90) eq 'Disabled') {
```

Comment:
I recommend to use proceed_on_failure => 0 here. "vmware-toolbox-cmd timesync status" should always return the actual status. If not, then there is a problem. 
So I think:
my $variable = script_output('vmware-toolbox-cmd timesync status', proceed_on_failure => 0, timeout => 90);
if ($variable eq 'Disabled')
is better.

Reply:
Actually, this is the command provided from open-vm-tools, and it will return '69' if its service status is 'Disabled'. So this is a specific return value, but we would not expect this command runs into a failure.

Reply:
I see. If I understand correctly, this test will enable timesync anyway if "the if" fails or true.

Reply:
Yes, that's right. We just want to match the string of the service status via this command.